integration-cli/docker_api_containers_test.go

```diff
@@ -879,11 +879,14 @@ func (s *DockerSuite) TestContainerApiBadPort(c *check.C) {
 	config := map[string]interface{}{
 		"Image": "busybox",
 		"Cmd":   []string{"/bin/sh", "-c", "echo test"},
-		"PortBindings": map[string]interface{}{
-			"8080/tcp": []map[string]interface{}{
-				{
-					"HostIP":   "",
-					"HostPort": "aa80",
+		"HostConfig": map[string]interface{}{
+			"NetworkMode": "default",
```

Comment:
Arguably the tests were incorrect before - they were passing an incomplete structure to the daemon as the network mode was validated on the client before. As it's being validated on the server now, it must be present.